typing/datarepr.ml

```diff
@@ -126,7 +121,7 @@ let constructor_descrs ~current_unit ty_path decl cstrs =
         let cstr_name = Ident.name cd_id in
         let existentials, cstr_args, cstr_inlined =
           let representation =
-            if decl.type_unboxed.unboxed
+            if rep = Variant_unboxed
             then Record_unboxed true
             else Record_inlined idx_nonconst
```

Comment:
I would prefer if this code matched on `rep`.

Reply:
Good point. (I was making fairly mechanical replacements here, didn't notice that this should really be a match)